### ISSUE TITLE
2.4.10 release

### ISSUE THREAD

**Issue by srh** [rethinkdb/rethinkdb-python#301]
With fixes for Python 3.10-3.12, this driver needs a 2.4.10 release.

- [x] Merge #297
- [x] Merge #296
- [x] Merge #298 (which depends on #296)
- [x] Merge #300 (which depends on #298)
- [x] Merge #302 (which depends on #300)
- [x] Merge #303
- [x] Double-check that import and export scripts basically work.  (And dump and restore.)
- [x] Fix #304
- [x] Update the build dependencies (see #306)
- [x] Fix #307

**Comment by lsabi** [comment]
@srh The last but one point has been merged into the `master` codebase. For reference [https://github.com/rethinkdb/rethinkdb-python/pull/303](https://github.com/rethinkdb/rethinkdb-python/pull/303)

The last point has been carried out on a separate branch a while ago from me. See [https://github.com/rethinkdb/rethinkdb-python/tree/lsabi/commands](https://github.com/rethinkdb/rethinkdb-python/tree/lsabi/commands). Since I started it, there have been some changes in the `master` branch, thus some adjustments shall be done. Furthermore, the commands were not complete and are probably missing something.

**Comment by srh** [comment]
@lsabi I was imprecise -- I meant the `rethinkdb dump` and `rethinkdb restore` commands.  I tried them out and they worked.  Also, they work by using the _import and _export modules that the `import` and `export` command line commands use directly.

**Comment by lsabi** [comment]
@srh no problem. Then everything is completed for now. I shall check my branch if it's worth keeping or not.

**Comment by srh** [comment]
My understanding is, that branch has stuff not in `master` that you wanted to develop.  I'd like to pop off this small release because there are Python 3.10-3.12 users who can't use all features of the driver right now.

**Comment by lsabi** [comment]
The idea was to refactor the entire codebase in order to make more readable and support the latest python versions. Unfortunately, time has been against us and couldn't finish it before new versions/features were released.

I need to figure out how the building automation works. @gabor-boros can you help?

**Comment by lsabi** [comment]
@gabor-boros I think I do not have the permissions to self-review my own PRs. Can you review it?

**Comment by fdemian** [comment]
Hi, everyone. 
I'm using this driver as well and having problems with async support and it seems like this new release would solve them.
Is there anything I can do to hep this issue move forward?

**Comment by gabor-boros** [comment]
Closing this PR as the new version is just released: https://pypi.org/project/rethinkdb/

**Comment by srh** [comment]
Thanks @gabor-boros !  I've added a v2.4.10 tag and given it a GitHub release page: https://github.com/rethinkdb/rethinkdb-python/releases/tag/v2.4.10

**Comment by gabor-boros** [comment]
Oh, I forgot to push the tag. Anyway, thanks @srh!
